packages/flutter/lib/src/rendering/object.dart

```diff
@@ -3718,6 +3718,11 @@ abstract class RenderObject with DiagnosticableTreeMixin implements HitTestTarge
   /// object, for accessibility purposes.
   Rect get semanticBounds;
 
+  /// Wether the semantic bounding box will be clipped by [parent].
+  ///
+  /// Defaults to true,
+  bool get semanticsBoundsClippedByParent => true;
```

Comment:
This only fixes for semantics rect. the tap target not only applies when screen reader is on, but also when people interact with the app directly. which means the gesture recognizer (and thus the renderobject) also needs to be above the tap target size

Reply:
Recording showing that the hit target is also fixed by this PR (`_DayPeriodInputPadding` was there before but its size was wrong):

https://github.com/user-attachments/assets/b6b61346-e1ee-4bcd-8fbd-91e5d12ecbe1